File: frontend/src/constants/contractType.ts
/**
 * 合同类型常量定义
 * 统一管理合同类型的英文代码和中文显示名称映射关系
 * 
 * @author CodeBuddy
 * @since 2025-08-15
 */

// 合同类型选项（用于下拉选择）
export const CONTRACT_TYPE_OPTIONS = [
  { value: 'RENT', label: '租赁' },
  { value: 'SHOP', label: '商铺' },
  { value: 'OFFICE', label: '办公' },
  { value: 'OFFICE_BUILDING', label: '写字楼' },
  { value: 'COWORKING', label: '联合办公' },
  { value: 'WAREHOUSE', label: '仓库' },
  { value: 'PARKING', label: '停车位' }
] as const

// 合同类型映射（用于显示转换）
export const CONTRACT_TYPE_MAP: Record<string, string> = {
  'RENT': '租赁',
  'SHOP': '商铺',
  'OFFICE': '办公',
  'OFFICE_BUILDING': '写字楼',
  'COWORKING': '联合办公',
  'WAREHOUSE': '仓库',
  'PARKING': '停车位'
}

// 合同类型标签颜色映射
export const CONTRACT_TYPE_TAG_MAP: Record<string, string> = {
  'RENT': 'primary',
  'SHOP': 'success',
  'OFFICE': 'info',
  'OFFICE_BUILDING': 'warning',
  'COWORKING': 'danger',
  'WAREHOUSE': 'default',
  'PARKING': 'processing'
}

// 合同类型枚举类型
export type ContractTypeCode = 'RENT' | 'SHOP' | 'OFFICE' | 'OFFICE_BUILDING' | 'COWORKING' | 'WAREHOUSE' | 'PARKING'

/**
 * 根据合同类型代码获取显示名称
 * @param code 合同类型代码
 * @returns 中文显示名称
 */
export const getContractTypeName = (code: string): string => {
  return CONTRACT_TYPE_MAP[code] || code
}

/**
 * 根据合同类型代码获取标签颜色
 * @param code 合同类型代码
 * @returns 标签颜色类型
 */
export const getContractTypeTag = (code: string): string => {
  return CONTRACT_TYPE_TAG_MAP[code] || 'default'
}

/**
 * 验证合同类型代码是否有效
 * @param code 合同类型代码
 * @returns 是否有效
 */
export const isValidContractType = (code: string): code is ContractTypeCode => {
  return Object.keys(CONTRACT_TYPE_MAP).includes(code)
}

File: backend/src/main/java/com/cysz/minimal/enums/ContractType.java
package com.cysz.minimal.enums;

/**
 * 合同类型枚举
 * 统一管理合同类型的英文代码和中文显示名称
 * 
 * @author CodeBuddy
 * @since 2025-08-15
 */
public enum ContractType {
    
    RENT("RENT", "租赁"),
    SHOP("SHOP", "商铺"),
    OFFICE("OFFICE", "办公"),
    OFFICE_BUILDING("OFFICE_BUILDING", "写字楼"),
    COWORKING("COWORKING", "联合办公"),
    WAREHOUSE("WAREHOUSE", "仓库"),
    PARKING("PARKING", "停车位");
    
    private final String code;
    private final String displayName;
    
    ContractType(String code, String displayName) {
        this.code = code;
        this.displayName = displayName;
    }
    
    public String getCode() {
        return code;
    }
    
    public String getDisplayName() {
        return displayName;
    }
    
    /**
     * 根据代码获取枚举
     */
    public static ContractType fromCode(String code) {
        if (code == null) {
            return null;
        }
        for (ContractType type : ContractType.values()) {
            if (type.code.equals(code)) {
                return type;
            }
        }
        throw new IllegalArgumentException("未知的合同类型代码: " + code);
    }
    
    /**
     * 根据代码获取显示名称
     */
    public static String getDisplayNameByCode(String code) {
        ContractType type = fromCode(code);
        return type != null ? type.displayName : code;
    }
    
    /**
     * 验证代码是否有效
     */
    public static boolean isValidCode(String code) {
        if (code == null) {
            return false;
        }
        for (ContractType type : ContractType.values()) {
            if (type.code.equals(code)) {
                return true;
            }
        }
        return false;
    }
    
    /**
     * 获取所有合同类型选项
     */
    public static ContractType[] getAllTypes() {
        return ContractType.values();
    }
}